pennylane/data/data_manager/__init__.py

```diff
@@ -331,15 +331,15 @@ def load(  # pylint: disable=too-many-arguments
     """
     params = format_params(**params)
 
+    if data_name == "other":
+        data_name = params[0]["values"][0]
```

Comment:
Is there a way we can validate attributes for `other`?

Reply:
"other" doesn't have attributes because it doesn't have a `"class"` (a pennylane-dataset-specific json definition of "class"). 

The attributes vary from item to item in the "other" category.

For example:
`qml.data.load('other', name='mqt-bench')`: attributes are `ae`, `qaoa`, etc.
`qml.data.load('other', name='max-3-sat')`: attributes are `clauses`, `hamiltonians`, etc.

We can retrieve the attributes of the datasets in 'other' and verify their attributes because they _do_ have classes. This is kind of what's happening here, the code says "forget about 'other' and use the next parameter".

It's a bit hacky, but I think refactoring/cleaning this code is out of scope for this PR.